src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -732,8 +732,48 @@ void runtime::gpu::GPU_Emitter::emit_Max(EMIT_ARGS)
     }
 
     const ngraph::op::Max* max = static_cast<const ngraph::op::Max*>(node);
-    auto& cudnn_emitter = external_function->get_primitive_emitter()->get_cudnn_emitter();
-    auto index = cudnn_emitter->build_primitive(max);
+
+    size_t index;
+    if (args[0].get_element_type() == element::i32)
+    {
+        // one of args0 axes has zero size, zero output, use args1 value
+        if (args[0].get_size() == 0)
+        {
+            writer << out[0].get_type()
+                   << " init_value = " << TypeInfo::Get(args[0].get_type())->min() << ";\n";
+            writer << "vector<" << out[0].get_type() << "> temp(" << out[0].get_size()
+                   << ", init_value);\n";
+            writer << "runtime::gpu::cuda_memcpyHtD(" << out[0].get_name()
+                   << ", (void*)temp.data(), " << out[0].get_size() << " * "
+                   << out[0].get_element_type().size() << ");\n";
+            return;
+        }
+        else if (args[0].get_size() == out[0].get_size())
+        {
+            kernel::emit_memcpyDtD(writer, out[0], args[0]);
+            return;
+        }
+        else
+        {
+            auto axes_set = max->get_reduction_axes();
```

Comment:
Rather than manually configuring from AxisSet -> AxisVector, let's just add a constructor to nvshape.hpp:
```
NVShape(const AxisVector& vec)
```
and then pass the reduction AxisSet directly to build_reduce.

Reply:
Done